src/Sylius/Behat/Context/Api/Shop/AddressContext.php

```diff
@@ -152,22 +169,26 @@ public function iDeleteTheAddress(string $fullName): void
      */
     public function iShouldHaveAddresses(int $count = 1): void
     {
-        $this->client->index();
+        $this->addressClient->index();
 
-        Assert::same(count($this->responseChecker->getCollection($this->client->getLastResponse())), $count);
+        Assert::same(count($this->responseChecker->getCollection($this->addressClient->getLastResponse())), $count);
     }
 
     /**
      * @Then this address should be assigned to :fullName
+     * @Then the address assigned to :fullName should be in my book
      */
     public function thisAddressShouldBeAssignedTo(string $fullName): void
     {
-        /** @var AddressInterface $address */
-        $address = $this->sharedStorage->get('address_assigned_to_' . $fullName);
-
-        $addressBook = $this->responseChecker->getCollection($this->client->getLastResponse());
-
-        Assert::true($this->addressBookHasAddress($addressBook, $address));
+        $addressIri = $this->getAddressIriFromAddressBookByFullName($fullName);
+        $addressBook = $this->responseChecker->getCollection($this->addressClient->getLastResponse());
+
+        Assert::notEmpty(
+            array_filter($addressBook, function($address) use ($addressIri) {
+            return $address['@id'] === $addressIri;
+        }),
+            sprintf('There is no Address assigned to %s', $fullName)
+        );
```

Comment:
I wonder if this method makes sense, because in `getAddressIriFromAddressBookByFullName` method you are getting an address from the address book and then you are checking if this address is in the same address book 😃 

Maybe you could use [an address transformer](https://github.com/Sylius/Sylius/blob/master/src/Sylius/Behat/Context/Transform/AddressContext.php#L154) to get an address by full name and then check if this address is in address book? Or even simple, just check if the value returned from `getAddressIriFromAddressBookByFullName` method is not null.